packages/runtime/test-runtime-utils/src/test/insecureUrlResolverTest.spec.ts

```diff
@@ -32,22 +33,45 @@ describe("Insecure Url Resolver Test", () => {
         assert(!!request.headers?.[CreateNewHeader.createNew],
             "Request should contain create new header");
         const url = `${hostUrl}?fileName=${fileName}`;
-        assert.equal(request.url, url, "Request url should match");
+        assert.strictEqual(request.url, url, "Request url should match");
     });
 
     it("Resolved CreateNew Request", async () => {
         const resolvedUrl = await resolver.resolve(request) as IFluidResolvedUrl;
         const documentUrl = `fluid://${new URL(ordererUrl).host}/${tenantId}/${fileName}`;
-        assert.equal(resolvedUrl.endpoints.ordererUrl, ordererUrl, "Orderer url should match");
-        assert.equal(resolvedUrl.url, documentUrl, "Document url should match");
+        assert.strictEqual(resolvedUrl.endpoints.ordererUrl, ordererUrl, "Orderer url should match");
+        assert.strictEqual(resolvedUrl.url, documentUrl, "Document url should match");
     });
 
     it("Test RequestUrl for a data store", async () => {
         const resolvedUrl = await resolver.resolve(request);
         const dataStoreId = "dataStore";
-        const response = await resolver.getAbsoluteUrl(resolvedUrl, dataStoreId);
+        const absoluteUrl = await resolver.getAbsoluteUrl(resolvedUrl, dataStoreId);
 
         const compUrl = `${hostUrl}/${tenantId}/${fileName}/${dataStoreId}`;
-        assert.equal(response, compUrl, "Url should match");
+        assert.strictEqual(absoluteUrl, compUrl, "Url should match");
+    });
+
+    it("Test RequestUrl for longer url", async () => {
+        const testRequest: IRequest = {
+            url: `https://localhost/${fileName}/random/random2`,
+            headers: {},
+        };
+        // Mocking window since the resolver depends on window.location.host
+        if (typeof window === "undefined" && typeof global === "object") {
+            // eslint-disable-next-line dot-notation
+            global["window"] = { location: { host:"localhost" } };
+        }
+
+        const resolvedUrl = await resolver.resolve(testRequest);
+        const dataStoreId = "dataStore";
+        const absoluteUrl = await resolver.getAbsoluteUrl(resolvedUrl, dataStoreId);
+        ensureFluidResolvedUrl(resolvedUrl);
+
+        const compResolvedUrl = `fluid://${new URL(ordererUrl).host}/${tenantId}/${fileName}/random/random2`;
```

Comment:
nit: Use dataObject or dataStore instead of comp because we don't use component anymore. So "dataObjectResolvedUrl" and "dataObjectResponseUrl"

Reply:
Yeah, testResolvedUrl is fine. Actually, you can use expectedResolvedUrl since this the expected value.